### ISSUE TITLE
TaskResultOption compatibility with TaskResult and friends

### ISSUE THREAD

**Issue by jozefRudy** [demystifyfp/FsToolkit.ErrorHandling#259]
**Is your feature request related to a problem? Please describe.**
We have multiple CEs, like Result, Option, Task, TaskResult, TaskResultOption.

I am relatively new to f#, but it seems to me that it is impractical to work with TaskResultOption CE, even though for single expression it's clearly nice. But if we have various levels of expressions, e.g. TaskResultOption followed by TaskResult/Result, and for them to be inside taskResultOption CE, we need to keep converting all following expressions to highest denominator (TaskResultOption).

here is a code snippet, where Mode.Parse is result and we need to convert it to TaskResultOption to be able to follow linearly and extract success result, same with LoadUniverseForStrategy, which is TaskResult (but not option):
```
            taskResultOption {
                let! strategy = this.LoadLatestStrategyQueryResult()
                let! mode = Mode.Parse strategy.StrategyType |> TaskResult.ofResult |> TaskResult.map Some
                let! universe = this.LoadUniverseForStrategy strategy.Id |> TaskResult.map Some

                return

                    { Id = strategy.Id
                      Entry = strategy.Entry
                      Exit = strategy.Exit
                      Saved = strategy.Saved
                      Universe = universe
                      Mode = mode }
            }
```

Is some kind of smart CE possible, that in case of using TaskResultOption would work using let! on TaskResult and friends?
Following snippet would be ideal:
```
            taskResultOption {
                let! strategy = this.LoadLatestStrategyQueryResult()
                let! mode = Mode.Parse strategy.StrategyType
                let! universe = this.LoadUniverseForStrategy strategy.Id
                return

                    { Id = strategy.Id
                      Entry = strategy.Entry
                      Exit = strategy.Exit
                      Saved = strategy.Saved
                      Universe = universe
                      Mode = mode }
            }
```

Maybe this is just me being a novice and not seeing a straightforward way forward.

**Comment by TheAngryByrd** [comment]
Yeah this particular CE doesn't have all the same niceness that all the others do. Any PRs would be appreciated here!

**Comment by jozefRudy** [comment]
not sure if i am up to a task. Also, maybe unrelated, `taskResult` cannot extract from simple `option` with let! Any advice on this?

**Comment by TheAngryByrd** [comment]
> taskResult cannot extract from simple option with let! Any advice on this?

Yeah `TaskResult` knows nothing about options. You'll have to convert the `Option` to a `Result`.

**Comment by jozefRudy** [comment]
yes, makes sense using `Result.requireSome` as an example, thanks
